components/bl-transferList-component/src/list/list-header.js

```diff
@@ -0,0 +1,41 @@
+// import { getToggleAllIcon } from '../helpers/get-toggle-all-icon';
+import { Checkbox } from './checkbox';
+
+export const ListHeader = props => {
+  const { items, title, selected, iconColor, setAllSelected } = props;
+
+  const isAllChecked = items.length > 0 && items.length === selected.length;
+  const isIndeterminated = selected.length > 0 && !isAllChecked;
+
+  const handleToggleAll = ({ target: { checked } }) => {
+    if (checked) {
+      setAllSelected(prevState => {
+        const selectedItems = items.filter(item => !prevState.includes(item));
```

Comment:
complexity is `O(n2)`, would be better to prepare a map and then check a particular item on the map and as a result, the complexity will be `O(n)`

```
const selectedItemsMap = prevState.reduce((m, item) => ({...m, [item]:1}), {})

items.filter(item => !selectedItemsMap[item]);
```

Reply:
Unfortunately, we cannot do this, because the checkboxes will work differently, and not as it was originally intended.

Reply:
do you mean this one won't work?

```
const handleToggleAll = ({ target: { checked } }) => {
    if (checked) {
      setAllSelected(prevState => {
        const prevStateMap = prevState.reduce((m, item) => ({...m, [item]:1}), {}) // it can be moved into a separate function
        const selectedItems = items.filter(item => !prevStateMap[item]);

        return prevState.concat(selectedItems);
      });
    } else {
      setAllSelected(prevState => {         
        const prevStateMap = prevState.reduce((m, item) => ({...m, [item]:1}), {}) // in order to reuse it here

         return prevState.filter(item => !prevStateMap[item])
    });
    }
  };

```

Reply:
Yes, it won't work as originally intended